### PR TITLE
Saving youtube url

### PR DESCRIPTION
**Pull request by Rafi993**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [X] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
It allows user to add their youtube url to their dev.to profile

## Related Tickets & Documents
https://github.com/thepracticaldev/dev.to/issues/5973

## Mobile & Desktop Screenshots/Recordings (if there are UI changes)

youtube_url
youtube_url_in_admin_view

## Added tests?

- [X] yes
- [ ] no, because they aren't needed
- [ ] no, because I need help

## Added to documentation?

- [ ] docs.dev.to
- [ ] readme
- [X] no documentation needed

## What gif best describes this PR or how it makes you feel?

Happy

### CODE REVIEW

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -328,6 +328,20 @@ def user_from_authorization_service(service_name, signed_in_resource, cta_varian
...
 
+    describe "#youtube_url" do
+      it "accepts valid https youtube url", :aggregate_failures do
+        %w[thepracticaldev thepracticaldev/ the.practical.dev].each do |username|
+          user.youtube_url = "https://youtube.com/#{username}"
+          expect(user).to be_valid
+        end
+      end
+
+      it "does not accept invalid facebook url" do
```
Comment (mstruve) [inline spec/models/user_spec.rb:339]: Probably want to update the description of this spec so there is no confusion 😉

Comment (Rafi993) [reply spec/models/user_spec.rb]: Awesome. Thanks for taking your time to review the PR.

**Comment by benhalpern** [comment]
Thanks for the PR! At a glance it looks great. I'll note that in the future it's best to ask if the feature is approved if it's not marked `ready for dev` in the issue. But that's just a little housekeeping note. We definitely approve of this!

So it'll still pend a technical review but we should be able to merge in not too long if things check out.

File: app/views/users/_profile.html.erb [hunk 2]
```diff
@@ -12,176 +12,181 @@
...
-    <div class="crayons-field">
-      <%= f.label :email, class: "crayons-field__label" %>
-      <%= f.text_field :email, class: "crayons-textfield", placeholder: "john.doe@example.com" %>
-    </div>
-
-    <div class="crayons-field crayons-field--checkbox">
-      <%= f.check_box :email_public, class: "crayons-checkbox" %>
-      <%= f.label :email_public, "Display email on profile", class: "crayons-field__label" %>
-    </div>
+<div class="crayons-card mb-6 grid grid-flow-row gap-6 p-6">
```
[the remark this answers is not in the document]
Comment (Rafi993) [reply app/views/users/_profile.html.erb:15]: Thanks @Ridhwana I ran svgo now.

File: app/assets/images/youtube-logo.svg [hunk 3]
```diff
@@ -0,0 +1,83 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
```
[the remark this answers is not in the document]
Comment (Rafi993) [reply app/assets/images/youtube-logo.svg]: Thanks @Ridhwana I ran svgo now.

[the remark this answers is not in the document]
Comment (Ridhwana) [reply app/assets/images/youtube-logo.svg]: Cool thanks, just the indentation and then this looks ready to go 👍

File: app/views/users/_profile.html.erb (hunk 2 above)
[the remark this answers is not in the document]
Comment (Rafi993) [reply app/views/users/_profile.html.erb]: I have made this change too. I think my prettier was messing it up.

**Comment by Zhao-Andy** [comment]
Merged! Thank you for your first contribution! 🎉 🎉 🎉

**Comment by Rafi993** [comment]
There was a bug created by running svgo over youtube-logo.svg It is resolved by PR https://github.com/thepracticaldev/dev.to/pull/7559

**Pull request merged**